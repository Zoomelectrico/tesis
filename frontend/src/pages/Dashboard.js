import React, { Suspense, lazy } from "react";
import { BrowserRouter as Router } from "react-router-dom";
import { Container } from "reactstrap";
import {
  Sidebar,
  NavbarAdmin,
  FooterAdmin,
  Loading,
  ProtectedRoute
} from "../components";

const DashHome = lazy(() => import("./dashboard/Home"));
const DashPostulate = lazy(() => import("./dashboard/Postulate"));
const DashProfile = lazy(() => import("./dashboard/Profile"));
const DashVote = lazy(() => import("./dashboard/Vote"));
const DashResults = lazy(() => import("./dashboard/Results"));

const routes = [
  {
    name: "Home",
    path: "/app/dashboard",
    icon: "fas fa-home",
    exact: true,
    component: props => <DashHome {...props} />
  },
  {
    name: "Profile",
    path: "/app/dashboard/profile",
    icon: "fas fa-user",
    exact: true,
    component: props => <DashProfile {...props} />
  },
  {
    name: "Postulate",
    path: "/app/dashboard/postulate",
    icon: "fas fa-user-plus",
    exact: true,
    component: props => <DashPostulate {...props} />
  },
  {
    name: "Vote",
    path: "/app/dashboard/vote",
    icon: "fas fa-receipt",
    exact: true,
    component: props => <DashVote {...props} />
  },
  {
    name: "Results",
    path: "/app/dashboard/results",
    icon: "fas fa-poll",
    exact: true,
    component: props => <DashResults {...props} />
  }
];

const Dashboard = props => (
  <Router>
    <Suspense fallback={<Loading />}>
      <Sidebar
        {...props}
        bgColor=""
        routes={routes}
        logo={{
          link: "/",
          src: require("../assets/img/logo-1-x2.png"),
          alt: "logo"
        }}
      />
      <div className="main-content">
        <NavbarAdmin {...props} brandText="brand" />
        {routes.map(({ name, path, component, exact }) => (
          <ProtectedRoute
            key={name}
            path={path}
            component={component}
            exact={exact}
          />
        ))}
        <Container fluid>
          <FooterAdmin />
        </Container>
      </div>
    </Suspense>
  </Router>
);

export default Dashboard;
