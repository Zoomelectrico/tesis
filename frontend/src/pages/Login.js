import React, { useEffect } from "react";
import {
  Row,
  Col,
  Container,
  Card,
  CardHeader,
  CardBody,
  FormGroup,
  Label,
  Input,
  Button
} from "reactstrap";
import { Navbar } from "../components";

const Login = props => {
  useEffect(() => {
    document.body.classList.add("bg-default");
    return function cleanup() {
      document.body.classList.remove("bg-default");
    };
  }, []);
  return (
    <div className="main-content">
      <Navbar {...props} />
      <div className="header bg-gradient-info py-7 py-lg-8">
        <Container>
          <Row className="justify-content-center">
            <Col sm="12" md="7" lg="5">
              <Card style={{ backgroundColor: "#f5f7f9" }}>
                <CardHeader>
                  <h3>Iniciar Sesion</h3>
                </CardHeader>
                <CardBody className="p-3">
                  <Row>
                    <Col md="12">
                      <FormGroup>
                        <Label for="email">Correo Electronico</Label>
                        <Input
                          onChange={props.onChangeLogin}
                          className="form-control-alternative"
                          type="email"
                          name="email"
                          id="email"
                          placeholder="email@correo.unimet.edu.ve"
                        />
                      </FormGroup>
                    </Col>
                    <Col md="12">
                      <FormGroup>
                        <Label for="password">Contrasena</Label>
                        <Input
                          onChange={props.onChangeLogin}
                          className="form-control-alternative"
                          type="password"
                          name="password"
                          id="password"
                          placeholder="********"
                        />
                      </FormGroup>
                    </Col>
                    <Col md="12 d-flex justify-content-center">
                      <Button
                        color="success"
                        className="my-auto"
                        onClick={props.login}
                      >
                        Iniciar Sesion
                      </Button>
                    </Col>
                  </Row>
                </CardBody>
              </Card>
            </Col>
          </Row>
        </Container>
      </div>
    </div>
  );
};

export default Login;
