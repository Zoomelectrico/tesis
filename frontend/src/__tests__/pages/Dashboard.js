import React from 'react';
import { shallow } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import Dashboard from '../../pages/Dashboard';
import Login from '../../pages/Login';

const props = {
  user: {
    firstName: 'Juan',
    privilege: 2,
  },
  update: jest.fn(),
  onChangeUpdate: jest.fn(),
};

describe('Dashboad', () => {
  it('Render Correctly', () => {
    const wrapper = shallow(
      <MemoryRouter initialEntries={['/app/dashboard']}>
        <Dashboard {...props} />
      </MemoryRouter>
    );
    expect(wrapper).toMatchSnapshot();
  });
  it('Redirect to Login', () => {
    const wrapper = shallow(
      <MemoryRouter initialEntries={['/app/dashboard']}>
        <Dashboard />
      </MemoryRouter>
    );
    console.log(wrapper);
  });
});
