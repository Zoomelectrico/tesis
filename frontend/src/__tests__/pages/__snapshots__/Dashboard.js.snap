// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboad Render Correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MemoryRouter
    initialEntries={
      Array [
        "/app/dashboard",
      ]
    }
  >
    <Dashboard
      onChangeUpdate={[MockFunction]}
      update={[MockFunction]}
      user={
        Object {
          "firstName": "Juan",
          "privilege": 2,
        }
      }
    />
  </MemoryRouter>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Dashboard
        onChangeUpdate={[MockFunction]}
        update={[MockFunction]}
        user={
          Object {
            "firstName": "Juan",
            "privilege": 2,
          }
        }
      />,
      "history": Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "9p49o3",
            "pathname": "/app/dashboard",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "9p49o3",
          "pathname": "/app/dashboard",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "onChangeUpdate": [MockFunction],
        "update": [MockFunction],
        "user": Object {
          "firstName": "Juan",
          "privilege": 2,
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Dashboard
          onChangeUpdate={[MockFunction]}
          update={[MockFunction]}
          user={
            Object {
              "firstName": "Juan",
              "privilege": 2,
            }
          }
        />,
        "history": Object {
          "action": "POP",
          "block": [Function],
          "canGo": [Function],
          "createHref": [Function],
          "entries": Array [
            Object {
              "hash": "",
              "key": "9p49o3",
              "pathname": "/app/dashboard",
              "search": "",
              "state": undefined,
            },
          ],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "index": 0,
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "key": "9p49o3",
            "pathname": "/app/dashboard",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "onChangeUpdate": [MockFunction],
          "update": [MockFunction],
          "user": Object {
            "firstName": "Juan",
            "privilege": 2,
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
